specification/liftrqumulo/resource-manager/Qumulo.Storage/preview/2022-06-27-preview/Qumulo.Storage.json

```diff
@@ -0,0 +1,753 @@
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/FileSystemResource.fileSystemName"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Resource deleted successfully."
+          },
+          "202": {
+            "description": "Resource deletion accepted."
+          },
+          "204": {
+            "description": "Resource already deleted."
+          },
+          "default": {
+            "description": "Common error response for all Azure Resource Manager APIs to return error details for failed operations.",
+            "schema": {
+              "$ref": "../../../../../common-types/resource-management/v3/types.json#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-long-running-operation-options": {
+          "final-state-via": "azure-async-operation"
+        },
+        "x-ms-long-running-operation": true,
+        "x-ms-examples": {
+          "FileSystems_Delete_MaximumSet_Gen": {
+            "$ref": "./examples/FileSystems_Delete_MaximumSet_Gen.json"
+          },
+          "FileSystems_Delete_MinimumSet_Gen": {
+            "$ref": "./examples/FileSystems_Delete_MinimumSet_Gen.json"
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "CompanyDetails": {
+      "type": "object",
+      "properties": {
+        "companyName": {
+          "type": "string",
+          "description": "Company name"
+        },
+        "officeAddress": {
+          "type": "string",
+          "description": "Office Address"
+        },
+        "countryOrRegion": {
+          "type": "string",
+          "description": "Country or region of operation"
+        },
+        "domain": {
+          "type": "string",
+          "description": "Domain of the company"
+        },
+        "business": {
+          "type": "string",
+          "description": "Business of the company"
+        },
+        "numberOfEmployees": {
+          "type": "integer",
+          "format": "int32",
+          "description": "Number of Employees"
+        }
+      },
+      "description": "Company details of Qumulo FileSystem resource",
+      "required": [
+        "companyName"
+      ]
+    },
+    "FileSystemResource": {
+      "type": "object",
+      "properties": {
+        "properties": {
+          "$ref": "#/definitions/FileSystemResourceProperties",
+          "description": "The resource-specific properties for this resource.",
+          "x-ms-mutability": [
+            "read",
+            "create"
+          ],
```

Comment:
FileSystemResourceProperties is mentioned here with "x-ms-mutability": ["read", "create"] -> means FileSystemResourceProperties can only be created during first put/create, however in Patch also, input parameters seem to be updating these props.

Can you please confirm?

Reply:
Yes, currently Patch is updating these properties. we have a work item to restrict Patch, updating these properties.

Reply:
So, what I understand is, Patch would treat this property as read-only, and return 200 (if matches with existing value) or 400 (if doesn't match with existing value). Correct?

Reply:
Yes, that's correct.